index.html

```diff
@@ -2053,34 +2041,33 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
     according to the algorithmic steps described in this specification.
   </section>
 
-  <section><h3>Handling Window visibility</h3>
+  <section><h3>Handling visibility change</h3>
     <p>
-      Each {{Window}} object where the Web NFC API is
-      exposed has separate <a>NFCWriter</a> and <a>NFCReader</a> instances.
-      The <a>visible</a> state of the
-      the <a>top-level browsing context</a>'s {{Document}} determines the
-      <a href="#nfc-suspended">suspended</a> state of the associated
-      <a>NFCWriter</a> and <a>NFCReader</a> instances.
+      When the user agent determines that the
+      <dfn data-cite="PAGE-VISIBILITY#dfn-visibility-states">visibility
+      state</dfn> of the [=environment settings object / responsible document=]
+      of the <a>current settings object</a> changes, it must run these steps:
     </p>
+    <ol class="algorithm">
+      <li>Let |document:Document| be the [=environment settings object /
+      responsible document=] of the <a>current settings object</a>.
+      </li>
+      <li>
+        If |document|'s <a>visibility state</a> is `"visible"`,
+        <a>resume NFC</a> and abort these steps.
+      </li>
+      <li>
+        Otherwise, <a>suspend NFC</a> and attempt to <a>abort a pending
+        push operation</a>.
+      </li>
```

Comment:
I wonder wouldn't be enough to suspend NFC, meaning that pending push will not fulfill even on taps? Imagine a visibility change for any reason and the user needs to set up another push. Just wondering should we keep the pending pushes and set up readers and only suspend NFC while they are not visible, instead of resetting the NFC related operations.

Reply:
We should not. It would be very weird to have backgrounded an app and 5 min you foreground it again and put the phone down on a tag and it pushes. I think aborting makes sense and also that way the app can handle it

Reply:
When foregrounding, it could list those current ops (that have not timeouted) and user has a choice to tap or not, or to delete them.

But maybe apps can handle these themselves. However, any system notification or anything that causes visibility change will reset the user flow, and that's intrusive.

Reply:
notifications shouldn't change visibility unless you actually click on them and they focus another app. This is fine. @riju

Reply:
We really dont want to implement such a thing, that is very complicated and requires a lot of UX discussions on Android and is basically a no-go

Reply:
OK, let's keep it simple and if any doubts, discuss in an issue.